File: 400 Nth Digit/nth_digit.py
class Solution:
    def findNthDigit(self, n: int) -> int:
        num_len = 1
        num_count = 9
        start = 1
        while (n > num_len * num_count):
            n -= num_len * num_count
            num_len += 1
            num_count *= 10
            start *= 10
            
        curr = start + (n - 1) / num_len
        place = (n - 1) % num_len
        return int(str(curr)[place])

File: 0400 Nth Digit/NthDigit.java
class Solution {
    public int findNthDigit(int n) {
        int len = 1;
        long count = 9;
        // first number of this length
        int start = 1;
    
        while (n > len * count) {
            n -= len * count;
            len++;
            count *= 10;
            start *= 10;
        }
        
        // find the number n is in
        int curr = start +  (n - 1) / len;
        int place = (n - 1) % len;
        return Character.getNumericValue(Integer.toString(curr).charAt(place));
    }
}